### PR TITLE
Add IPEX sentence transformers support 

### PR DESCRIPTION
**Pull request by echarlaix**
Will be updated after the IPEX refactorization is merged https://github.com/huggingface/optimum-intel/pull/1009 

cc @rbrugaro @yao-matrix @IlyasMoutawwakil @tomaarsen

### CODE REVIEW

File: tests/ipex/utils_tests.py [hunk 1]
```diff
@@ -53,6 +53,8 @@
     "resnet": "hf-internal-testing/tiny-random-resnet",
     "roberta": "hf-internal-testing/tiny-random-roberta",
     "roformer": "hf-internal-testing/tiny-random-roformer",
+    "sentence-transformers-bert": "sentence-transformers-testing/stsb-bert-tiny-safetensors",
```
Comment (IlyasMoutawwakil) [inline tests/ipex/utils_tests.py:56]:
```suggestion
    "st-bert": "sentence-transformers-testing/stsb-bert-tiny-safetensors",
```

File: tests/ipex/test_modeling.py [hunk 2]
```diff
@@ -510,3 +514,33 @@ def test_patched_model(self):
             transformers_outputs = transformers_model(**inputs)
         outputs = ipex_model(**inputs)
         self.assertTrue(torch.allclose(outputs.logits, transformers_outputs.logits, atol=1e-4))
+
+
+class IPEXSTModel(unittest.TestCase):
+    SUPPORTED_ARCHITECTURES = (
+        "sentence-transformers-bert",
```
Comment (IlyasMoutawwakil) [inline tests/ipex/test_modeling.py:521]:
```suggestion
        "st-bert",
```

**Review by IlyasMoutawwakil — APPROVED** [review]
LGTM !

File: optimum/intel/__init__.py [hunk 3]
```diff
@@ -55,6 +62,15 @@
         "IPEXModel",
     ]
 
+try:
+    if not (is_ipex_available() and is_sentence_transformers_available()):
+        raise OptionalDependencyNotAvailable()
+except OptionalDependencyNotAvailable:
+    _import_structure["utils.dummy_ipex_objects"].extend(["IPEXSentenceTransformer"])
```
Comment (ilya-lavrenov) [inline optimum/intel/__init__.py:69]: it broke OpenVINO path 
https://github.com/openvinotoolkit/openvino/actions/runs/12240792041/job/34146426674

**Pull request merged**